### ISSUE TITLE
outputs the word the even when nothing is said

### ISSUE THREAD

**Issue by tinker1234** [alphacep/vosk-api#797]
outputs the word the even when nothing is said

this is literally what it does

```bash
the
the
the pokemon the
the
the
the
the
the
the
the
the
the
```

**Comment by nshmyrev** [comment]
Bad, really bad! How can we help you?

**Comment by Cyborgscode** [comment]
Fedora 34: Pythoon 3.9 : big german model 0.21: vosk 0.3.32 : x86_64

Device: USB logitech webcam ( with a very sensetive mic )

similar, but different problem with german:

ghost words, no one said out loud, not in the fore- or background, appear from time to time. "Einen" is a common, but not exclusive one. i.e. "essen","jetzt","gibt", "schwimmen","können" , just to name some i just saw right now, while absolute silence was in the room.

Helpfull would be a tool, to calibrate the microphone input level, so that noise is not too loud to produce this.

**Comment by nshmyrev** [comment]
We need a recorded audiofile to analyze the problem. The microphone demo has an option to store the audio.

**Comment by Cyborgscode** [comment]
the raw recording (no idea if it is saved as wav or flac ) of this exact session is attached:

LOG (VoskAPI:CompileLooped():nnet-compile-looped.cc:345) Spent 0.00568414 seconds in looped compilation.
{
"text" : "nun"
}
{
"text" : "einen"
}
{
"text" : "einen"
}
{
"text" : "einen"
}
^C
Done
[vosk.test.recording.zip](https://github.com/alphacep/vosk-api/files/7764098/vosk.test.recording.zip)

**Comment by nshmyrev** [comment]
Great, thank you. I will take a look but it will take some time. Not an easy problem to solve.

**Comment by xenotropic** [comment]
Possibly related to [nerd-dictation #31](https://github.com/ideasman42/nerd-dictation/issues/31)("Large model surrounding sentences with 'the'"). I'm not the reporter of that issue but I can confirm. Seemed to be only with vosk-model-en-us-0.22 -- prior big model didn't do this, small model doesn't do it. Also can confirm that if I just sit with an open silent mic I get "the"s, but it's most noticeable that the "the"s get added before/after spoken text. Don't have time this moment to make a wav file that reproduces but will see if I can later.

**Comment by Cyborgscode** [comment]
one finding form observation and audio processings: 

some sort of "compression" audio filtering is done, which expands the volume of silent passages to make them analyseable. 
So the "silent mic" is no longer silent. A running tv show in a neighboring room repeatedly triggered the voice recognition, while a human observer would not recognize anything while watching the "background noise flatline" in the audacity recording of this triggering. In other words, after we filtered the audacity recording with the compression filter, we noticed, that there is valid talk to understand, if increased enough, and that is enough to trigger the recognition software. It does not understand everything, but from time to time, it gives out trigger keywords :)

Solution: add a configureable VOX level like in Teamspeak and ignore background noise. 

I suggested a simple mic testtool, which could manage this and several other audio recording related stuff:

- vox level
- volume increase factor 
- recognition test ( print text, let person talk and check response for adjusting the input level automatically to the perfect level )
etc.
